phoenix-core/src/main/java/org/apache/phoenix/coprocessor/DropColumnMutator.java

```diff
@@ -181,13 +183,28 @@ public MetaDataMutationResult validateAndAddMetadata(PTable table,
             byte[] key = mutation.getRow();
             int pkCount = getVarChars(key, rowKeyMetaData);
             if (isView && mutation instanceof Put) {
-                PColumn column = MetaDataUtil.getColumn(pkCount, rowKeyMetaData, table);
+                PColumn column = null;
+                // checking put from the view or index
+                if (Bytes.compareTo(schemaName, rowKeyMetaData[SCHEMA_NAME_INDEX]) == 0
+                        && Bytes.compareTo(tableName, rowKeyMetaData[TABLE_NAME_INDEX]) == 0) {
+                    column = MetaDataUtil.getColumn(pkCount, rowKeyMetaData, table);
+                } else {
+                    for(int i = 0; i < table.getIndexes().size(); i++) {
+                        PTableImpl indexTable = (PTableImpl) table.getIndexes().get(i);
+                        byte[] indexTableName = indexTable.getTableName().getBytes();
+                        byte[] indexSchema = indexTable.getSchemaName().getBytes();
+                        if (Bytes.compareTo(indexSchema, rowKeyMetaData[SCHEMA_NAME_INDEX]) == 0
+                                && Bytes.compareTo(indexTableName, rowKeyMetaData[TABLE_NAME_INDEX]) == 0) {
+                            column = MetaDataUtil.getColumn(pkCount, rowKeyMetaData, indexTable);
+                            break;
+                        }
+                    }
+                }
                 if (column == null)
                     continue;
                 // ignore any puts that modify the ordinal positions of columns
```

Comment:
Do we know why we are ignoring puts that modify ordinal positions?
FYI @twdsilva

Reply:
I think this was to handle older clients that update ordinal column positions when a column is dropped.